nlp/tests.py

```diff
@@ -15,3 +15,51 @@ def test_strip_spaces(self):
         result = preprocess_text(text, lemmatize=False)
         expected = (['test', 'test'], 'test test')
         self.assertEqual(result, expected)
+
+
+class GSTTestCase(TestCase):
+    def test_single_full_match(self):
+        text_a = "hello"
+        text_b = "how delightful, hello there"
+
+        expected = ([{'fromA': 0, 'fromB': 16, 'toA': 5, 'toB': 21}], 5)
+        result = greedy_string_tiling(text_a, text_b, len(text_a))
+
+        self.assertEqual(result, expected)
+
+    def test_single_partial_match(self):
+        text_a = b"hello"
+        text_b = b"we are in helsinki now"
+        match_len = 3
+
+        expected = ([{'fromA': 0, 'toA': 3, 'fromB': 10, 'toB': 13}], 3)
+        result = greedy_string_tiling(text_a, text_b, match_len)
+
+        self.assertEqual(result, expected)
+
+    def test_no_match(self):
+        text_a = b"hello"
+        text_b = b"go away, you nuisance"
+
```

Comment:
Binary strings are here to test whether it works with them as well?

Reply:
didn't mean to keep it here, i will make it normal string. but it works with them as well